server/plugin/command.go

```diff
@@ -323,3 +314,28 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 	p.postCommandResponse(args, fmt.Sprintf("Unknown action %v", action))
 	return &model.CommandResponse{}, nil
 }
+
+// parseCommand parses the entire command input string and retrieves the command, action and parameters
+func parseCommand(input string) (command, action string, parameters []string) {
+	// do not split if the whitespace is in between 2 quotes, e.g. "Hello World"
+	foundQuote := false
+	split := strings.FieldsFunc(input, func(c rune) bool {
```

Comment:
This relies on FieldsFunc processing the string in order. Taking a look to the documentation

>  FieldsFunc splits the string s at each run of Unicode code points c satisfying f(c) and returns an array of slices of s. If all code points in s satisfy f(c) or the string is empty, an empty slice is returned. **FieldsFunc makes no guarantees about the order in which it calls f(c)**. If f does not return consistent results for a given c, FieldsFunc may crash.

Sadly, we may have to "reimplement" FieldsFunc to make sure this will work. Thoughts, @hanzei ?

Reply:
@larkox Good catch, I didn't realize that. It's weird that the tests never failed so far though. I updated the PR with a custom implementation of the parser algorithm.

Reply:
Probably it hasn't failed because the go implementation does indeed always run in the same order. But since the specification is that way, they can potentially change that in other go implementations / versions. Therefore, better safe than sorry.